lib/kramdown/converter/html.rb

```diff
@@ -147,6 +147,14 @@ def convert_header(el, indent)
         if @options[:auto_ids] && !attr['id']
           attr['id'] = generate_id(el.options[:raw_text])
         end
+
+        if @options[:auto_ids] && @options[:header_links]
+          link = Element.new(:a, nil, nil)
+          link.attr['href'] = "##{attr['id']}"
+          link.children << el.children.pop
```

Comment:
A header element may have more than one children, e.g. `# This *is* something`.

Reply:
And I would *not* make the whole header itself clickable since that would potentially interfere with a link in the header itself. Why not just generate an empty `<a>` tag at the beginning of the header and let the user style that to their liking? E.g. by showing some content when hovering over the header.